tokenauth/token.go

```diff
@@ -93,6 +112,24 @@ func (t *TokenAuth) CheckToken(token string, purpose string) (*Claims, error) {
 	return claims, nil
 }
 
+func (t *TokenAuth) retryCheckToken(token string, purpose string) (*Claims, error) {
+	retryClaims, retryErr := t.CheckToken(token, purpose)
+	if retryErr != nil {
+		t.blacklistLock.Lock()
+		if len(t.blacklist) >= t.blacklistSize {
+			t.blacklist[t.blacklistHead] = token
+			t.blacklistHead += 1
+			if t.blacklistHead >= t.blacklistSize {
+				t.blacklistHead = 0
+			}
+		} else {
+			t.blacklist = append(t.blacklist, token)
+		}
```

Comment:
I think the order of the queue is not preserved with this approach. When the queue fills to maximum capacity, the newest element should still be at the end of the queue rather than the beginning. So, when the queue reaches maximum length, I think we should do `t.blacklist = t.blacklist[1:]`, then append the newest element as always.